CONTRIBUTING.md

```diff
@@ -144,6 +145,23 @@ second line is always blank, and other lines should be wrapped at 80 characters.
 This allows the message to be easier to read on GitHub as well as in various
 git tools.
 
+### Commit History
+To prepare your branch to open a PR, you will need to have the minimal number of logical commits so we can maintain
+a clean commit history. Most commonly a PR will include a single commit where all changes are squashed, although
+sometimes there will be multiple logical commits.
+
+```bash
```

Comment:
It might be useful to remove this snippet and just link to a guide on squashing commits that we deem trustworthy. Maybe like this one [here](https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history).

Reply:
Sounds good. Do you think adding a link from [`git-scm.com`](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) would be more reasonable?

Reply:
Definitely, good call!